src/DataTypes/DataTypeDynamic.h

```diff
@@ -0,0 +1,53 @@
+#pragma once
+
+#include <DataTypes/IDataType.h>
+
+#define DEFAULT_MAX_DYNAMIC_TYPES 32
+
+
+namespace DB
+{
+
+class DataTypeDynamic final : public IDataType
+{
+public:
+    static constexpr bool is_parametric = true;
+
+    DataTypeDynamic(size_t max_dynamic_types_ = DEFAULT_MAX_DYNAMIC_TYPES);
+
+    TypeIndex getTypeId() const override { return TypeIndex::Dynamic; }
+    const char * getFamilyName() const override { return "Dynamic"; }
+
+    bool isParametric() const override { return true; }
+    bool canBeInsideNullable() const override { return false; }
+    bool supportsSparseSerialization() const override { return false; }
```

Comment:
this indicates sub column in Dynamic can't use sparse serialization?

Reply:
By now yes. And I am not sure that sparse serialization is needed for Dynamic subcolumns, Dynamic stores Variant column inside and in Variant column for each variant type we store only rows with this variant (so no default values on rows with other variants and it's kind of already in sparse form).